### PR TITLE
Blueshield station trait

### PR DESCRIPTION
**Pull request by Swiftfeather**
## About The Pull Request

Puts blueshield behind a new station trait, much like bridge assistant or cargorilla. 

List of changes I have made: 

- Blueshield can now only be played if the station trait rolls or is bussed.
- Blueshield lockers will no longer be spawned on the map. (I think, I'm not a mapper lmao)
- Blueshields will spawn with a beacon in their backpack that will supply drop a blueshield locker when used. 
- When the Blueshield trait rolls, it will be available for latejoins to choose. 

Weighting is subject to change. 

_**Also maintainers the config needs to be updated to accommodate the slot # change (I think???)**_

## Why It's Good For The Game

I have had a feeling for a while when playing this server that Blueshield is often poorly utilized. This isn't a dig at any specific players, the role just feels like it doesn't fit, especially when I'm playing command, and ESPECIALLY when there's an NT rep present. 

Originally I wanted to remove it but after playing a bit more I felt that was going too far. 

I think putting it behind a trait will improve the enjoyment both the people who play and the people who interact with because it will be more sought after and I believe people who see it open and pick it will try to make the most out of their shift.

Admins can force a slot open or bus the station trait to take effect next-round if they've got plans and want to make it available, or someone's been very good and redeemed their good bubber points.

## Proof Of Testing

No blueshield without the trait.
image

I bussed the trait then restarted the server, blueshield available for latejoins.
image

Blueshield cannot be selected in the character preferences occupations tab.
image

Spawned with the beacon and used it.
image

Pod came down and dropped the BS locker off
image

## Changelog

:cl:
add: Added the Blueshield station trait.
del: Blueshields are no longer available every round.
del: Blueshield lockers have been removed from the maps.
add: Blueshields spawn with a beacon to supply drop their locker in.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/blueshield/code/blueshield.dm [hunk 1]
```diff
@@ -53,6 +53,11 @@
 	ears = /obj/item/radio/headset/headset_bs/alt
 	glasses = /obj/item/clothing/glasses/hud/security/sunglasses
 	implants = list(/obj/item/implant/mindshield)
+	//BUBBER ADDITION START
+	backpack_contents = list(
+		/obj/item/choice_beacon/blueshield = 1,
+	)
+	//BUBBER ADDITION END
```
Comment (Majkl-J) [inline modular_skyrat/modules/blueshield/code/blueshield.dm:60]: This one can come in modular form

```suggestion
/datum/outfit/job/blueshield
	backpack_contents = list(
		/obj/item/choice_beacon/blueshield = 1,
	)
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/blueshield/code/blueshield.dm [hunk 2]
```diff
@@ -4,8 +4,8 @@
 	auto_deadmin_role_flags = DEADMIN_POSITION_SECURITY
 	department_head = list(JOB_NT_REP)
 	faction = FACTION_STATION
-	total_positions = 1
-	spawn_positions = 1
+	total_positions = 0 //BUBBER EDIT Original = 1
+	spawn_positions = 0 //BUBBER EDIT Original = 1
```
Comment (Majkl-J) [inline modular_skyrat/modules/blueshield/code/blueshield.dm:8]: Same here
```suggestion
/datum/job/blueshield
	total_positions = 0
	spawn_positions = 0
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]

File: modular_skyrat/modules/blueshield/code/blueshield.dm (hunk 1 above)
Comment (Swiftfeather) [reply modular_skyrat/modules/blueshield/code/blueshield.dm]: committed it modularly

File: modular_skyrat/modules/blueshield/code/blueshield.dm (hunk 2 above)
Comment (Swiftfeather) [reply modular_skyrat/modules/blueshield/code/blueshield.dm]: committed it modularly

**Comment by nevimer** [comment]
not sure im sold on it

**Comment by Swiftfeather** [comment]
> not sure im sold on it

It feels like most of the time blueshield is played for the extended access and better weapons and gear, not so much to be a bodyguard to the heads of staff. When they do focus on guarding it's often just the captain; one of the most well equipped people on the station, going to check on other heads only when they are hurt or don't have maxed sensors. 

This makes antag roleplaying with captains pretty much impossible because the blueshield will have their gun out before the other person has even finished typing their first message. It makes it more difficult to do stuff with the rest of the heads too. When red alerts come on I've seen BS suddenly procure an entire armory and medbay from their bag they've been stockpiling the whole shift. 

Blueshields were added at a time when heads were constantly being annihilated, mostly because Skyrat was far more hugboxy at that point. I don't see the same issues here. 

NT rep feels far better in comparison as far as roleplaying goes with command, it makes sense, while Blueshield feels stuck back in the arms race gunsec era.

**Comment by AllCorruptingWyrm** [comment]
I'm against Blueshield being touched, in fact I just made new regulations for the job that need to get parsed and approved.

**Comment by projectkepler-ru** [comment]
I'll look over this in the coming days.

**Comment by Erol509** [comment]
Blueshield is very popular role, and people like it. Im aginst this PR.

**Comment by Sylphet** [comment]
I like blueshield, this isn't great.

**Comment by projectkepler-ru** [comment]
My say is final on this, it will be addressed later with an update to the role itself in a follow up PR and blueshield oversight.

**Comment by ReturnToZender** [comment]
While I don't like this PR, I _do_ sorta like the Blueshield having a beacon to drop their locker and equipment. It's just cool.

**Pull request closed without merging**